packages/x-charts/src/RadarChart/RadarDataProvider/RadarDataProvider.tsx

```diff
@@ -0,0 +1,219 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { MakeOptional } from '@mui/x-internals/types';
+import { ChartContainerProps } from '../../ChartContainer';
+import { RadarSeriesType } from '../../models/seriesType/radar';
+import { AxisConfig, ChartsRadiusAxisProps, ChartsRotationAxisProps } from '../../models/axis';
+import { ChartDataProvider } from '../../ChartDataProvider';
+import {
+  useChartPolarAxis,
+  UseChartPolarAxisSignature,
+} from '../../internals/plugins/featurePlugins/useChartPolarAxis';
+import {
+  useChartHighlight,
+  UseChartHighlightSignature,
+} from '../../internals/plugins/featurePlugins/useChartHighlight';
+import { radarSeriesConfig } from '../seriesConfig';
+import { DEFAULT_MARGINS } from '../../constants';
+import { RadarConfig } from './radar.types';
+
+const RADAR_SERIES_CONFIG = { radar: radarSeriesConfig };
+const RADAR_PLUGINS = [useChartPolarAxis, useChartHighlight] as const;
+
+type RadarPluginSignatures = [UseChartPolarAxisSignature, UseChartHighlightSignature];
+
+export interface RadarDataProviderProps
+  extends Omit<
+    ChartContainerProps<'radar', RadarPluginSignatures>,
+    'series' | 'plugins' | 'rotationAxis' | 'radiusAxis' | 'dataset'
+  > {
+  /**
+   * The series to display in the bar chart.
+   * An array of [[RadarSeriesType]] objects.
+   */
+  series: MakeOptional<RadarSeriesType, 'type'>[];
+  /**
+   * The configuration of the radar scales.
+   */
+  radar: RadarConfig;
+}
```

Comment:
I like the API, sounds reasonable. `series` handles the series, `radar` is the chart config 👍